src/AutoRest.CSharp/MgmtTest/Generation/Mock/MgmtMockTestBaseWriter.cs

```diff
@@ -211,27 +284,85 @@ protected void WriteTestOperation(CodeWriterDeclaration declaration, MockTestCas
             // we will always use the Async version of methods
             if (testCase.ClientOperation.IsPagingOperation)
             {
-                _writer.Append($"await foreach (var _ in ");
+                _writer.Append($"{(generatingSample ? $"" : $"await ")}foreach (var _ in ");
                 WriteTestMethodInvocation(declaration, testCase);
                 using (_writer.Scope($")"))
-                { }
+                {
+                    if (generatingSample && !testCase.ClientOperation.ReturnType.IsFrameworkType && testCase.ClientOperation.ReturnType.Implementation is Resource)
+                    {
+                        _writer.UseNamespace("System.Text.Json");
+                        _writer.Line($"Console.WriteLine($\"ResourceId: {{_.Data.Id}}\");");
+                        _writer.Line($"Console.WriteLine($\"Full Resource content: {{JsonSerializer.Serialize(_.Data)}}\");");
```

Comment:
I do not think using the JsonSerializer to serialize the resource data here is a good idea, because our serializer is designed for the request body, therefore the readonly properties will not be printed out. This must introduce a big confusion to our customers if they just copy this sample somewhere else.
We could consider to write another helper method like `Print{ResourceName}` which writes the properties one by one from that resource data